toolchain/lc-compile/src/set.c

```diff
@@ -22,7 +22,7 @@ typedef unsigned int SetRef;
 
 #define MAX_SET_DEPTH 256
 SetRef s_sets[MAX_SET_DEPTH];
-unsigned int s_set_count = 0;
```

Comment:
@montegoulding If you look at the set API used in this file, you'll see that it uses `int` parameters throughout (for example, see `EnsureSetStackDepth()`.  Fairly often, this results in an `int` being compared with `s_set_count`.

When comparing an `int` to an `unsigned int`, the `int` is converted to the equivalent unsigned representation, bitwise.  For example, consider:

```C++
int a = -1;
unsigned int b = 1;
bool c = (a < b);
```

In this case, `c = false` because -1 maps to `UINT_MAX`, which is more than 1.

There were two options here:

1. I could refactor the whole file to add overflow checks everywhere in order to ensure correctness.
2. I could change `s_set_count` to `int`